optimade/server/data/__init__.py

```diff
@@ -15,6 +15,9 @@
     try:
         with open(Path(__file__).parent / path) as f:
             globals()[var] = bson.json_util.loads(f.read())
+
+        if var == "structures":
+            globals()[var] = sorted(globals()[var], key=lambda x: x["task_id"])
```

Comment:
It could also be fixed by applying a default sort to the MongoDB collection, but I'm not sure if that is generally desirable, so I prefer this way of doing it.